### ISSUE TITLE
Unable to run the code for Jumping-task

### ISSUE THREAD

**Issue by dibbla** [google-research/google-research#1163]
hi @agarwl

Thanks for releasing the code. However, I have some trouble when attempting to run the jumping-task code. I've downloaded the code under this [dir](https://github.com/google-research/google-research/tree/master/pse), and I've followed the README instruction and run `python -m jumping_task.train --train_dir . --training_epochs 1` under my `/pse` dir. 

Here is what I got:
```
Traceback (most recent call last):
  File "/home/dibbla/miniconda3/envs/pse/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/dibbla/miniconda3/envs/pse/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/dibbla/RLgen/paperCode/pse/jumping_task/train.py", line 417, in <module>
    app.run(main)
  File "/home/dibbla/miniconda3/envs/pse/lib/python3.8/site-packages/absl/app.py", line 312, in run
    _run_main(main, args)
  File "/home/dibbla/miniconda3/envs/pse/lib/python3.8/site-packages/absl/app.py", line 258, in _run_main
    sys.exit(main(argv))
  File "/home/dibbla/RLgen/paperCode/pse/jumping_task/train.py", line 411, in main
    train_agent(train_dir, measurements)
  File "/home/dibbla/RLgen/paperCode/pse/jumping_task/train.py", line 186, in train_agent
    imitation_data = data_helpers.generate_imitation_data(
  File "/home/dibbla/RLgen/paperCode/pse/jumping_task/data_helpers.py", line 108, in generate_imitation_data
    imitation_data[color_key] = _generate_imitation_data(
  File "/home/dibbla/RLgen/paperCode/pse/jumping_task/data_helpers.py", line 87, in _generate_imitation_data
    observations, actions, rewards = generate_optimal_trajectory(
  File "/home/dibbla/RLgen/paperCode/pse/jumping_task/data_helpers.py", line 53, in generate_optimal_trajectory
    initial_obs = env.environment._reset(  # pylint: disable=protected-access
  File "/home/dibbla/miniconda3/envs/pse/lib/python3.8/site-packages/gym/core.py", line 235, in __getattr__
    raise AttributeError(
AttributeError: attempted to get missing private attribute '_reset'
```

It seems like accessing a protected method in gym/core. So I downgraded the gym to 0.21. The problem stays still. Here is my environment:

- Ubuntu 20.04LTS
- RTX3090
- gym 0.21.0
- gym-jumping-task 0.0.1 (from anothor [repo](https://github.com/Maluuba/jumping-task))
- tensorflow 2.4.1

And I also downloaded all requirements in requirements.txt.

Any clue about solving the problem?

**Comment by dibbla** [comment]
Well, seems like I'm not really calling the _reset for gym-jumping-task

**Comment by agarwl** [comment]
Btw, you want to use the jumping task repo: https://github.com/google-research/jumping-task as the original one didn't have extra support for some of the experiments in the paper.

**Comment by agarwl** [comment]
So the `_reset` method needs to be called like it was called earlier i suppose: `env.environment._reset` (also, the `reset` function internally uses `_reset`: https://github.com/google-research/jumping-task/blob/master/gym_jumping_task/envs/jumping_task.py#L210)

**Comment by dibbla** [comment]
Sorry for the delay. 
It seems like the problem is 'TimeLimit' Wrapper. Without the wrapper the env is good with `_reset`. So I changed [here](https://github.com/google-research/google-research/blob/4808a726f4b126ea38d49cdd152a6bb5d42efdf0/pse/jumping_task/data_helpers.py#L53) with 
`initial_obs = env.environment.env._reset(  # pylint: disable=protected-access
      obstacle_position=obstacle_position,
      floor_height=floor_height)`

This `_reset` works. But it still reported error that I shoud take a reset before step. So I added `env.environment.reset()` before the previous changed line. It seems to work well. Maybe it is a proper change that matches your idea?

**Comment by dibbla** [comment]
As it produces the similar results in the paper, I believe this is a good way to run the code. Also thank you, Agarwl :smile:

**Comment by 6xa1v348** [comment]
Hi @dibbla and @agarwl, I'm not sure if i am doing it right but i'm not getting anything remotely close to what was presented in the paper. The only changes that I made was to include the `env.environment.reset()`. The evaluation grid for the 'wide' configuration from using `seed=1` looks like this
![image](https://user-images.githubusercontent.com/59470295/233958324-06a3136d-15e5-4f30-8f99-63f3a9feb025.png)
One issue that I noticed is that the `alignment_loss` does not seem to be dropping much, starting from approximately 8 and dropping quickly to 6 before fluctuating around this range until the end of training. For comparison, the `cross_entropy_loss` is in the range of `1e-6`.
The evaluation grid for `random grid` looks equally off. (Also using the `seed=1`) with multiple training environments not solved in the final trained model.
![image](https://user-images.githubusercontent.com/59470295/233961670-3fc2d1e0-24f2-48d7-89db-b73f0f700d35.png)

It will be great if you guys could point out any error in my runs.

**Comment by dibbla** [comment]
> Hi @dibbla and @agarwl, I'm not sure if i am doing it right but i'm not getting anything remotely close to what was presented in the paper. The only changes that I made was to include the `env.environment.reset()`. The evaluation grid for the `wide` configuration from using `seed=1` looks like this ![image](https://user-images.githubusercontent.com/59470295/233958324-06a3136d-15e5-4f30-8f99-63f3a9feb025.png) One issue that I noticed is that the `alignment_loss` does not seem to be dropping much, starting from approximately 8 and dropping quickly to 6 before fluctuating around this range until the end of training. For comparison, the `cross_entropy_loss` is in the range of `1e-6`. The evaluation grid for `random grid` looks equally off. (Also using the `seed=1`) with multiple training environments not solved in the final trained model. ![image](https://user-images.githubusercontent.com/59470295/233961670-3fc2d1e0-24f2-48d7-89db-b73f0f700d35.png)
> 
> It will be great if you guys could point out any error in my runs. Thanks!

It's been a while and I am no longer working on the pse🤣.

By the change you made, do you mean the one discusses in this issue? If you didn't encouter any bug, maybe you can try to reproduce the result with the very original code.

**Comment by 6xa1v348** [comment]
Thanks for the quick reply!
I only made the changes were the ones discussed in this issue (namely the `reset()` function for `gym` to work) and I'm running the code as is. I did follow the hyperparameters mentioned in the paper appendix but right now, it doesn't seem to work well. I will try to play around with the hyperparameters and coefficients to see if I can get something close to the paper!

**Comment by 6xa1v348** [comment]
Just to clarify, I was able to reproduce results similar to those presented in the paper after downgrading the environments to those given above by @dibbla (namely `gym==0.21.0` and `tensorflow==2.4.1`). I'm not too familiar with tensorflow so the problems I faced previously may be possibly due to the difference in version. The evaluation grid for `wide` using `seed=1` is 
![image](https://user-images.githubusercontent.com/59470295/235869249-d5f4cb96-409c-4e83-89d2-887cfb8da3a0.png)

Thanks!
